solutions/silver/cf-1744F.mdx

```diff
@@ -0,0 +1,144 @@
+---
+id: cf-1744F
+source: CF
+title: MEX vs MED
+author: Alex Du
+---
+
+## Explanation
+
+[Official Analysis (No Code)](https://codeforces.com/blog/entry/108101)
+
+<Spoiler title="Hint 1">
+
+How do we know if a specific subsegment satisfies the constraints for a given
+$\texttt{mex}$?
+
+First, to reach that value of $\texttt{mex}$, we must find subsegments which contain all
+integers $0, 1, \ldots, \texttt{mex}-1$, while also not containing $\texttt{mex}$.
+
+Then, given a subsegment, $\texttt{mex} > \texttt{med}$ will be satisfied if the length of the
+subsegment is $\leq 2 \cdot \texttt{mex}$, since the median will be at most $\texttt{mex}-1$.
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+Combining the two ideas in Hint 1, we can first find $l$ and $r$, which represent the
+indices of the left-most and right-most occurences of the numbers
+$0, 1, \ldots \texttt{mex} - 1$ respectively. Then, find all subsegments that contain the
+subsegment $l \ldots r$ within themselves, have length $\leq 2 \cdot \texttt{mex}$, and
+do not contain $\texttt{mex}$ itself. Now, we just need to count these
+subsegments efficiently.
+
+</Spoiler>
+
+<Spoiler title="Solution">
+
+Let $\texttt{pos}$ be the index of $\texttt{mex}$. If we know $l$, $r$, and $\texttt{pos}$, then we can
+count the subsegments for this $\texttt{mex}$ quickly.
+
+First, if $\texttt{pos}$ is between $l$ and $r$, the number of subsegments with this
+$\texttt{mex}$ is zero, since the $\texttt{mex}$ of this subsegment is greater than the $\texttt{mex}$ we
+are looking for.
+
+We need to find all subsegments that contain $l \ldots r$, that don't contain
+$\texttt{pos}$. Say $\texttt{pos}>r$, since we only need to slightly change our approach for
+$\texttt{pos} < l$. We can increment $r$ from $r$ to $\texttt{pos}-1$ fast enough, since each time
+we increment we increase the size of our subsegment by $1$, for a maximum of $N$
+times.
```

Comment:
```suggestion
Let $\texttt{pos}$ be the index of $\texttt{mex}$. If we know $l$, $r$, and $\texttt{pos}$, then we can quickly count the subsegments for this $\texttt{mex}$.
		

First, if $\texttt{pos}$ is between $l$ and $r$, the number of subsegments with this $\texttt{mex}$ is zero since the $\texttt{mex}$ of this subsegment is greater than the $\texttt{mex}$ we are looking for.
		

We need to find all subsegments that contain $l \ldots r$ that don't include $\texttt{pos}$. Say $\texttt{pos}>r$, since we only need to change our approach slightly $\texttt{pos} < l$, we can increment $r$ from $r$ to $\texttt{pos}-1$ fast enough, since each time we increment we increase the size of our subsegment by $1$, for a maximum of $N$ times.
```